src/wrover_status_light/ServiceIntro.md

```diff
@@ -0,0 +1,160 @@
+```
+rospy.spin()
+```
+Keeps the code running by entering a loop that listens for incoming requests. 
+
+The function `example_service_execution()` is the handler function for this service and is called whenever the node receives a new request message. Handler functions are responsible for carrying out the actual logics of the service and send a response back to the client. Handler functions always take in a single argument of the request message type, and returns an instance of the response message type.
+
+```
+return ExamplePackage.srv.ExampleServiceResponse(0.0)
+```
+In this example, the handler function always returns a hardcoded response of 0.0. If the handler function wants to use the request inputs, they can be accessed with `request.input1` or `request.input2`.
+
+Returning response messages can also be streamlined by not having to manually create a new instance. In this case, simply 
+```
+return 0.0
+``` 
+would also work. 
+
+### C++
+A server node can be initialized with
+```
+bool ExampleServiceServer::callback(ExmaplePackage::ExampleService::Request& request, ExmaplePackage::ExampleService::Response& response) {
+    return 0.0;
+}
+
+ExmaplePackage::ExampleService exampleService;
+ros::NodeHandle nh;
+ros::ServiceServer service = nh.advertiseService("example_service", &ExampleServiceServer::callblack, &exampleService);
+```
+To break the code down, the function `ExampleServiceServer::callback()` is the handler function of the server, and is ran every time the server receives a new request. The handler function must take in two arguments, a request and a response.
+```
+ExmaplePackage::ExampleService exampleService;
+ros::NodeHandle nh;
+ros::ServiceServer service = nh.advertiseService("example_service", &ExampleServiceServer::callblack, &exampleService);
+```
+The code sets up the service server using a `NodeHandle` instance. NodeHandles provide functions for creating, managing, and interacting using ROS nodes. Here the `NodeHandle` creates a new server with `example_service` as the server name, `ExmapleServiceServer::callback` as the handler function for service calls, and `exmapleService` defines teh service type. 
+
+## Writing a Client Node
+### Python
+Services can be called by 
+```
+rospy.wait_for_service('example_service')
+example_service = rospy.ServiceProxy('example_service', ExamplePackage.srv.ExampleService)
+response = example_service(ExamplePackage.srv.ExampleServiceRequest(..., ...))
+```
+To break the code down,
+```
+rospy.wait_for_service('exmaple_service')
+```
+This is a blocking operation that waits until a service with the name `example_service` is created. 
+```
+example_service = rospy.ServiceProxy('example_service', ExamplePackage.srv.ExampleService)
+```
+Creates a new callable proxy object for the service named `example_service` and defined by `ExmaplePackage.srv.ExampleService`.
```

Comment:
```suggestion
Creates a new callable proxy object for the service named `example_service` and defined by `ExamplePackage.srv.ExampleService`.
```

Reply:
I think "new callable proxy object" might be difficult for beginners to follow. Maybe just "new object" is enough here.